packages/init/src/SliceMachineInitProcess.ts

```diff
@@ -0,0 +1,1114 @@
+								)}: you are not a developer or admin of this repository`,
+							);
+						}
+					} else {
+						const validation = await validateRepositoryDomainAndAvailability({
+							domain,
+							existsFn: (domain) =>
+								this.manager.prismicRepository.checkExists({ domain }),
+						});
+						const errorMessage = getErrorMessageForRepositoryDomainValidation({
+							validation,
+							displayDomain: chalk.cyan(domain),
+						});
+
+						if (errorMessage) {
+							throw new Error(errorMessage);
+						}
+					}
+
+					task.title = `Selected repository ${chalk.cyan(
+						domain,
+					)} (flag ${chalk.cyan("repository")} used)`;
+
+					this.context.repository = {
+						domain,
+						exists: !!maybeRepository,
+					};
+				},
+			},
+		]);
+	}
+
+	protected async selectRepository(): Promise<void> {
+		assertExists(
+			this.context.userRepositories,
+			"User repositories must be available through context to run `selectRepository`",
+		);
+
+		if (this.context.userRepositories.length) {
+			await this.trySelectExistingRepository();
+		}
+
+		if (!this.context.repository) {
+			await this.selectNewRepository();
+		}
+
+		assertExists(
+			this.context.repository,
+			"Repository selection must be available through context to proceed",
+		);
+		// We prefer to manually allow console logs despite the app being a CLI to catch wild/unwanted console logs better
+		// eslint-disable-next-line no-console
+		console.log(
+			`${logSymbols.success} Selected repository ${chalk.cyan(
+				this.context.repository.domain,
+			)}`,
+		);
+	}
+
+	protected async trySelectExistingRepository(): Promise<void> {
+		assertExists(
+			this.context.userRepositories,
+			"User repositories must be available through context to run `trySelectExistingRepository`",
+		);
+
+		const { maybeDomain } = await prompt<string, "maybeDomain">({
+			type: "select",
+			name: "maybeDomain",
+			message: "Pick a repository to connect to or choose to create a new one",
+			warn: "You are not a developer or admin of this repository",
+			choices: [
+				{
+					title: "CREATE NEW",
+					description: "Create a new Prismic repository\n",
+					value: "",
+				},
+				...this.context.userRepositories
+					.map((repository) => {
+						const hasWriteAccess =
+							this.manager.prismicRepository.hasWriteAccess(repository);
+
+						return {
+							title: `${repository.domain}${
+								hasWriteAccess ? "" : " (Unauthorized)"
+							}`,
+							description: `Connect to ${chalk.cyan(repository.domain)}`,
+							value: repository.domain,
+							disabled: !hasWriteAccess,
+						};
+					})
+					.sort((a, b) => (a.value > b.value ? 1 : -1)),
+			],
+		});
+
+		if (maybeDomain) {
+			this.context.repository = {
+				domain: maybeDomain,
+				exists: true,
+			};
+		}
+	}
+
+	protected async selectNewRepository(): Promise<void> {
+		let suggestedName = "";
+
+		// TODO: Improve concurrency
+		const trySuggestName = async (name: string): Promise<string | void> => {
```

Comment:
Do we need to suggest a name?
maybe ask the design team about this :)

Reply:
It's designed to help developers start a new project quickly. Picking a repository name can be a time consuming process, so by suggesting a name, we make it easier for developers to skip that step and move forward.

The downside of this feature is that users cannot rename their repositories after creation. If they use one of the suggested names, they are stuck with it until they create a new repository.

I agree with you; we should consult the design/product team to confirm this is the behavior we want. I think we do. 🙂